[18.0][FIX] sale_order_line_date: fix setting empty commitment date on stock moves

StefanRijnhart opened the pull request:
Port of https://github.com/OCA/sale-workflow/pull/3364

Fixes #3363:
```
TypeError: unsupported operand type(s) for -: 'datetime.datetime' and 'NoneType'
```
in stock.move's _set_date_deadline

sebalix commented on sale_order_line_date/models/sale_order_line.py:
```diff
@@ -30,9 +30,19 @@ def _prepare_procurement_values(self, group_id=False):
         return vals
 
     def write(self, vals):
+        """Propagate a new commitment date to pending stock moves"""
```
It's not encouraged to override the docstring of such method, as it hides the original docstring in some IDE.
Better to put this as comment.

StefanRijnhart replied:
Thanks, fixed!

sebalix reviewed (approved):
LG, small comment

metaminux committed:
[FIX] sale_order_line_date: fix setting empty commitment date on stock moves

Fixes #3363:
```
TypeError: unsupported operand type(s) for -: 'datetime.datetime' and 'NoneType'
```
in stock.move's _set_date_deadline

sebalix commented:
BTW I spotted also this one done in 14.0: https://github.com/OCA/sale-workflow/pull/1902
It also updates the schedule date, didn't check into the details.

For now we can merge this one as it's coming from 16.0!

/ocabot merge patch

StefanRijnhart commented:
@sebalix That one looks a lot like https://github.com/OCA/sale-workflow/commit/fd06e8fb8da014e7772916cd30ebc63b68b9d45a which is included in 18.0.
